Migrations/20241201161752_trainingInit.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace mPole.Migrations
+{
+    /// <inheritdoc />
+    public partial class trainingInit : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "TrainingId",
+                schema: "dbo",
+                table: "Move",
+                type: "int",
+                nullable: true);
+
+            migrationBuilder.CreateTable(
+                name: "Training",
+                schema: "dbo",
+                columns: table => new
+                {
+                    Id = table.Column<int>(type: "int", nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    Name = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Description = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Type = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Duration = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Date = table.Column<DateTime>(type: "datetime2", nullable: false),
+                    Location = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Trainer = table.Column<string>(type: "nvarchar(max)", nullable: false),
```

Comment:
Majster, dawanie wszystkich kolumn jako nvarchar(max) jest słabe wydajnościowo, warto ustalać długość stringów i zasady albo w ModelBuilder w klasie kontekstu, albo wpiąć EntityConfiguration jakiś -


 `using Elastic.DigitalHQ.Domain.EntityModel;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace Elastic.Orchard.Modules.DigitalHQ.EntityConfiguration
{
    public class ZoneConfiguration : IEntityTypeConfiguration<ZoneEntity>
    {
        public void Configure(EntityTypeBuilder<ZoneEntity> builder)
        {
            builder.ToTable("Zone");

            builder.HasKey(l => l.Id)
                .IsClustered()
                .HasName("PK_Zone_Id");

            builder.Property(l => l.ExternalId)
                .IsRequired()
                .HasDefaultValueSql("NEWID()");

            builder.HasIndex(l => l.ExternalId)
                .IsClustered(clustered: false)
                .HasDatabaseName("UQ_Zone_ExternalId");

            builder.Property(l => l.Name)
                .IsRequired()
                .HasColumnType("nvarchar(100)");

            builder.Property(l => l.Description)
                .IsRequired(false)
                .HasDefaultValue(null)
                .HasColumnType("nvarchar(4000)");

            builder.Property(l => l.Coordinates)
                .IsRequired(false)
                .HasDefaultValue(null)
                .HasColumnType("nvarchar(4000)");

            builder.Property(l => l.CreatedOn)
                .HasDefaultValueSql("GETUTCDATE()")
                .ValueGeneratedOnAdd()
                .IsRequired();

            builder.Property(l => l.ModifiedOn);

            builder.Property(l => l.OrganizationId)
                .IsRequired();

            builder.Property(l => l.IsDeleted)
                .HasDefaultValue(false);

            builder.Property(l => l.DeletedAt)
                .HasDefaultValue(null);

            builder.Property(l => l.FloorId)
                .IsRequired();

            builder.HasOne(l => l.Floor)
                .WithMany()
                .HasForeignKey(l => l.FloorId)
                .OnDelete(DeleteBehavior.NoAction);
        }
    }
}
` - https://codeburst.io/ientitytypeconfiguration-t-in-entityframework-core-3fe7abc5ee7a?gi=77dca094ab7e